File: genguid.py
#!/usr/bin/python3
import random
random.seed()
b=[]
i=0
while i<16:
  r=round(random.random()*255)
  b.append(r)
  i+=1
b[6]&=0xf
b[6]^=0x40
b[8]&=0x3f
b[8]^=0x80
out=""
i=0
while i<4:
  out += '%02x' % b[i]
  i+=1
i=0
while i<3:
  f=0
  out+="-"
  while f<2:
    out+= "%02x" % b[i+4+f]
    f+=1
  i+=1
out+="-"
i=10
while i<16:
  out+="%02x" % b[i]
  i+=1
print(out)

File: genguid.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
    int i,j;
    time_t t;
    unsigned char bytes[16];

    srand((unsigned)time(&t));

    for(i = 0; i < 16; i++)
        bytes[i] = (unsigned char)(rand() % 255);

    bytes[6] &= 0xf;
    bytes[6] ^= 0x40;
    bytes[8] &= 0x3f;
    bytes[8] ^= 0x80;

    for(i = 0; i < 4; i++)
        printf("%02x", bytes[i]);
    for(j = 0; j < 3; j++)
    {
        printf("-");
        for(i = 0; i < 2; i++)
            printf("%02x", bytes[4+i+j]);
    }
    printf("-");
    for(i = 10; i < 16; i++)
        printf("%02x", bytes[i]);
    printf("\n");
    return 0;
}
